test/unittests/eof_validation_test.cpp

```diff
@@ -677,55 +673,52 @@ TEST_F(eof_validation, deprecated_instructions)
+    add_test_case(
+        eof_bytecode(127 * push0() + callf(1) + OP_STOP, 127).code(OP_RETF, 127, 127, 127),
         EOFValidationError::success);
 
-    add_test_case("EF0001 010008 02000200010001 040000 00 00800000 80800080 00 E4",
+    add_test_case(eof_bytecode(OP_STOP, 0).code(OP_RETF, 128, 128, 128),
         EOFValidationError::inputs_outputs_num_above_limit);
 
-    add_test_case("EF0001 010008 020002000400FF 040000 00 0080007F 007F007F E3000100" +
-                      127 * bytecode{1} + OP_RETF,
+    add_test_case(
+        eof_bytecode(callf(1) + OP_STOP, 127).code(127 * bytecode{1} + OP_RETF, 0, 127, 127),
         EOFValidationError::success);
 
-
-    add_test_case(bytecode{"EF0001 010008 02000200010101 040000 00 00800000 00810081"} + OP_STOP +
-                      128 * bytecode{1} + OP_RETF,
+    add_test_case(eof_bytecode(OP_STOP).code(128 * bytecode{1} + OP_RETF, 0, 129, 129),
         EOFValidationError::inputs_outputs_num_above_limit);
 
-    add_test_case("EF0001 010008 02000200830080 040000 00 0080007F 7F00007F" +
-                      127 * bytecode{"5F"} + "E3000100" + 127 * OP_POP + OP_RETF,
+    add_test_case(eof_bytecode(127 * push0() + callf(1) + OP_STOP, 127)
+                      .code(127 * OP_POP + OP_RETF, 127, 0, 127),
         EOFValidationError::success);
 
-    add_test_case(bytecode{"EF0001 010008 02000200010081 040000 00 00800000 80000080"} + OP_STOP +
-                      128 * OP_POP + OP_RETF,
+    add_test_case(eof_bytecode(OP_STOP, 0).code(128 * OP_POP + OP_RETF, 128, 0, 128),
         EOFValidationError::inputs_outputs_num_above_limit);
 }
 
 TEST_F(eof_validation, max_stack_height)
 {
-    add_test_case("EF0001 010008 02000200040BFE 040000 00 00800000 000003FF E3000100" +
-                      0x3FF * bytecode{1} + 0x3FF * OP_POP + OP_RETF,
+    add_test_case(eof_bytecode(callf(1) + OP_STOP, 0)
+                      .code(0x3FF * bytecode{1} + 0x3FF * OP_POP + OP_RETF, 0, 0, 1023),
         EOFValidationError::success);
 
-    add_test_case("EF0001 010008 0200020C010001 040000 00 008003FF 00000000" + 0x3FF * bytecode{1} +
-                      0x3FF * OP_POP + "E3000100" + OP_RETF,
+    add_test_case(eof_bytecode(1023 * bytecode{1} + 1023 * OP_POP + callf(1) + OP_STOP, 1023)
+                      .code(OP_RETF, 0, 0, 0),
         EOFValidationError::success);
 
-    add_test_case(bytecode{"EF0001 010008 02000200010C01 040000 00 00800000 00000400"} + OP_STOP +
-                      0x400 * bytecode{1} + 0x400 * OP_POP + OP_RETF,
+    add_test_case(
+        eof_bytecode(OP_STOP).code(1024 * bytecode{1} + 1024 * OP_POP + OP_RETF, 0, 0, 1024),
         EOFValidationError::max_stack_height_above_limit);
 
-    add_test_case("EF0001 010008 0200020C010001 040000 00 00800400 00000000" + 0x400 * bytecode{1} +
-                      0x400 * OP_POP + OP_STOP + OP_RETF,
+    add_test_case(
+        eof_bytecode(0x400 * bytecode{1} + 0x400 * OP_POP + OP_STOP, 1024).code(OP_RETF, 0, 0, 0),
```

Comment:
this one I'm not really sure what's the point, but at least it misses callf
```suggestion
        eof_bytecode(0x400 * bytecode{1} + callf(1) + 0x400 * OP_POP + OP_STOP, 1024).code(OP_RETF, 0, 0, 0),
```